lib/src/piece/import.dart

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import '../back_end/code_writer.dart';
+import '../constants.dart';
+import 'piece.dart';
+
+/// An import or export directive.
```

Comment:
How about naming it "DirectivePiece"?

Reply:
Libraries are directives too. :-/ The Analyzer API calls the union of "import and export directive" a "namespace directive", but that feels a little... arbitrary.

The comment in [PieceFactory] is relevant here:

```dart
/// Naming these functions can be hard. For example, there isn't an obvious
/// word for "import or export directive" or "named thing with argument list".
/// To avoid that, we pick one concrete construct formatted by the function,
/// usually the most common, and name it after that, as in [createImport()].
```

I ran into this problem all the time with the old IR. Collection literals, function calls with trailing commas, and lambdas get similar formatting. What is a word that describes all of those? (I started using "block", but that word already means a couple of other things, so it gets confusing.) Or a word that describes an asserts and a function call?

So with the new one, the idea is that sometimes I'll just pick one member of the set of things that share formatting logic, call it canonical, and use that for the name. When I can, I try to pick a more general name, like "Sequence" or "Infix". But otherwise, I just pick a specific name and then document the set of things it applies to.

Reply:
Just thought of another option, what do you think about `UriDirectivePiece`? Library directives may have a name, but not a URI, and imports/exports both always have an URI.

> I ran into this problem all the time with the old IR. Collection literals, function calls with trailing commas, and lambdas get similar formatting. What is a word that describes all of those?

`DelimitedCommaBlockPiece`?

I do see the tension between descriptive and terse names - I think a name based an a representative example has an opportunity cost and it might be worth bikeshedding a bit.